src/chainstate/stacks/boot/pox-2.clar

```diff
@@ -0,0 +1,792 @@
+                      true)
+                  (err ERR_DELEGATION_EXPIRES_DURING_LOCK)))
+
+      ;; stacker principal must not be stacking
+      (asserts! (is-none (get-stacker-info stacker))
+        (err ERR_STACKING_ALREADY_STACKED))
+
+      ;; the Stacker must have sufficient unlocked funds
+      (asserts! (>= (stx-get-balance stacker) amount-ustx)
+        (err ERR_STACKING_INSUFFICIENT_FUNDS))
+
+      ;; ensure that stacking can be performed
+      (try! (minimal-can-stack-stx pox-addr amount-ustx first-reward-cycle lock-period))
+
+      ;; register the PoX address with the amount stacked via partial stacking
+      ;;   before it can be included in the reward set, this must be committed!
+      (add-pox-partial-stacked pox-addr first-reward-cycle lock-period amount-ustx)
+
+      ;; add stacker record
+      (map-set stacking-state
+        { stacker: stacker }
+        { amount-ustx: amount-ustx,
+          pox-addr: pox-addr,
+          first-reward-cycle: first-reward-cycle,
+          lock-period: lock-period })
+
+      ;; return the lock-up information, so the node can actually carry out the lock. 
+      (ok { stacker: stacker,
+            lock-amount: amount-ustx,
+            unlock-burn-height: unlock-burn-height })))
+
+;; Reject Stacking for this reward cycle.
+;; tx-sender votes all its uSTX for rejection.
+;; Note that unlike PoX, rejecting PoX does not lock the tx-sender's
+;; tokens.  PoX rejection acts like a coin vote.
+(define-public (reject-pox)
+    (let (
+        (balance (stx-get-balance tx-sender))
+        (vote-reward-cycle (+ u1 (current-pox-reward-cycle)))
+    )
+
+    ;; tx-sender principal must not have rejected in this upcoming reward cycle
+    (asserts! (is-none (get-pox-rejection tx-sender vote-reward-cycle))
+        (err ERR_STACKING_ALREADY_REJECTED))
+
+    ;; tx-sender can't be a stacker
+    (asserts! (is-none (get-stacker-info tx-sender))
+        (err ERR_STACKING_ALREADY_STACKED))
+
+    ;; vote for rejection
+    (map-set stacking-rejection
+        { reward-cycle: vote-reward-cycle }
+        { amount: (+ (next-cycle-rejection-votes) balance) }
+    )
+
+    ;; mark voted
+    (map-set stacking-rejectors
+        { stacker: tx-sender, reward-cycle: vote-reward-cycle }
+        { amount: balance }
+    )
+
+    (ok true))
+)
+
+;; Used for PoX parameters discovery
+(define-read-only (get-pox-info)
+    (ok {
+        min-amount-ustx: (get-stacking-minimum),
+        reward-cycle-id: (current-pox-reward-cycle),
+        prepare-cycle-length: (var-get pox-prepare-cycle-length),
+        first-burnchain-block-height: (var-get first-burnchain-block-height),
+        reward-cycle-length: (var-get pox-reward-cycle-length),
+        rejection-fraction: (var-get pox-rejection-fraction),
+        current-rejection-votes: (next-cycle-rejection-votes),
+        total-liquid-supply-ustx: stx-liquid-supply,
+    })
+)
+
+
+(define-public (stack-extend (extend-count uint)
```

Comment:
Can you add a method-level comment block to describe this new method, and indicate that it's a new change in 2.1?

Reply:
It probably makes more sense to document the method in the `docs` module, but I can add a marker here to indicate that it is new in 2.1.

Reply:
the other methods are documented in this file

Reply:
Added documentation for these methods in bbd826d4bde81aee6c26f134439b9d73d8885e34